### PR TITLE
On create, copy image data for named volumes.

### LINKED ISSUE #18670: Named volumes do not copy existing data from image path to volume.
When a container is created, a volume can be used which is mounted on a directory which already contains data from the image.

For anonymous volumes, this data is copied to the volume before mounting over the directory in the image.

However, for named volumes, this data is *not* copied. Even if the volume driver is local.

Here is the anonymous volume case:
````
$ docker run -it --rm -v /home busybox sh
````
````
DEBU[0088] Registering new volume reference: driver , name 9829aab88c7b64631f6472ecb97a18643258389d3027f90192c573f33e32c450 
DEBU[0088] Incrementing volume reference: driver local, name 9829aab88c7b64631f6472ecb97a18643258389d3027f90192c573f33e32c450 
DEBU[0088] Creating dest directory: /var/lib/docker/volumes/9829aab88c7b64631f6472ecb97a18643258389d3027f90192c573f33e32c450/_data 
DEBU[0088] Calling TarUntar(/var/lib/docker/devicemapper/mnt/5704f1232e1ef266ffba028f655d61cbbdc4aeb9726fd3f661d3f2c394d1ebe2/rootfs/home, /var/lib/docker/volumes/9829aab88c7b64631f6472ecb97a18643258389d3027f90192c573f33e32c450/_data) 
DEBU[0088] TarUntar(/var/lib/docker/devicemapper/mnt/5704f1232e1ef266ffba028f655d61cbbdc4aeb9726fd3f661d3f2c394d1ebe2/rootfs/home /var/lib/docker/volumes/9829aab88c7b64631f6472ecb97a18643258389d3027f90192c573f33e32c450/_data) 
````
````
# ls /home
default  ftp
````

And here is the named volume case:
````
$ docker run -it --rm -v myvol:/home busybox sh
````
````
DEBU[57755] Registering new volume reference: driver local, name myvol 
DEBU[57755] Incrementing volume reference: driver local, name myvol 
````
````
# ls /home
# 
````

So the question is: why is this behavior different? This behavior should not be different. And I'm sure this used to work in the past, perhaps in 1.8? Please take a look. Thanks!

It seems to me that **daemon.createContainerPlatformSpecificSettings()** is not even being called for the named volume case.

````
$ bundles/1.10.0-dev/dynbinary/docker version
Client:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.2
 Git commit:   75d69ce-dirty
 Built:        Mon Dec 14 23:27:34 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.10.0-dev
 API version:  1.22
 Go version:   go1.5.2
 Git commit:   75d69ce-dirty
 Built:        Mon Dec 14 23:27:34 2015
 OS/Arch:      linux/amd64
````

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes #18670

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM 🐭

**Comment by cpuguy83** [comment]
Pinging @thaJeztah, as I'm not sure where, if any, docs exist that would need to be changed around this. I think this is generally expected behavior given the docs for `docker run -v /foo`

**Comment by thaJeztah** [comment]
oh :heart_eyes: thanks @cpuguy83. hm good one, I'll have a look for possible docs changes. I'll add the impact/changelog label, because this may be something to mention :)

**Comment by crosbymichael** [comment]
There shouldn't need any docs changes because this is a regression from how existed volumes worked

File: daemon/create_unix.go [hunk 1]
```diff
@@ -60,14 +61,24 @@ func (daemon *Daemon) createContainerPlatformSpecificSettings(container *contain
 			return err
 		}
 
-		// never attempt to copy existing content in a container FS to a shared volume
-		if v.DriverName() == volume.DefaultDriverName {
```
Comment (crosbymichael) [inline daemon/create_unix.go]: Shouldn't this only happen for the default?

Comment (cpuguy83) [inline daemon/create_unix.go]: We talked about it on the review call and said it should happen for any driver.
I believe this part was to make sure this wasn't a bind? I'll have to pull up that PR again.

Comment (crosbymichael) [inline daemon/create_unix.go]: Yes, this shouldn't happen for bind mounts

Comment (cpuguy83) [inline daemon/create_unix.go]: It's not, see `populateVolumes` below, which is what replaced this call.
There's also a test (already part of the codebase) which ensures this.

**Comment by calavera** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Comment by thaJeztah** [comment]
I searched the docs, and I couldn't find a reference to "named" volumes not being propagated, so docs changes are indeed not needed.

**Pull request merged**